lib/rouge/lexers/sas.rb

```diff
@@ -0,0 +1,563 @@
+        rule /\b(_n_|_error_|_file_|_infile_|_msg_|_iorc_|_cmd_)\b/i, Name::Builtin::Pseudo
+
+        # auto variable list names
+        rule /\b(_character_|_numeric_|_all_)\b/i, Name::Builtin
+
+        # datalines/cards etc
+        rule /\b(datalines|cards)(\s*)(;)/i do
+          groups Keyword, Text, Punctuation
+          push :datalines
+        end
+        rule /\b(datalines4|cards4)(\s*)(;)/i do
+          groups Keyword, Text, Punctuation
+          push :datalines4
+        end
+
+
+        # operators (Language Reference p96)
+        rule %r(\*\*|[\*/\+-]), Operator
+        rule /[^¬~]?=:?|[<>]=?:?/, Operator
+        rule /\b(eq|ne|gt|lt|ge|le|in)\b/i, Operator::Word
+        rule /[&|!¦¬∘~]/, Operator
+        rule /\b(and|or|not)\b/i, Operator::Word
+        rule /[<>|><]/, Operator # min/max
+        rule /\|\|/, Operator # concatenation
+
+        # The OF operator should also be highlighted (Language Reference p49)
+        rule /\b(of)\b/i, Operator::Word
+        rule /\b(like)\b/i, Operator::Word # Language Ref p181
+      
+        rule /\d+/, Num::Integer
+
+        rule /\$/, Keyword::Type
+
+        # Macro definitions
+        rule /(%macro|%mend)(\s*)([\w\d]+)/i do
+          groups Keyword, Text, Name::Function
+        end
+        rule /%mend/, Keyword
+
+        rule /%\w[\w\d]*/ do |m|
+          if self.class.sas_macro_statements.include? m[0].upcase
+            token Keyword
+          elsif self.class.sas_macro_functions.include? m[0].upcase
+            token Keyword
+          else
+            token Name
+          end
+        end
+      end
+
+      state :basics2 do
+        # Rules to be parsed after the keywords (which are different depending
+        # on the context)
+
+        # Missing values (Language Reference p81)
+        rule /\s\.[;\s]/, Keyword::Constant # missing
+        rule /\s\.[a-z_]/, Name::Constant # user-defined missing
+
+        rule /[\(\),;:\{\}\[\]\\\.]/, Punctuation
+
+        rule /@/, Str::Symbol # line hold specifiers
+        rule /\?/, Str::Symbol # used for format modifiers
+        
+        rule /[^\s]+/, Text # Fallback for anything we haven't matched so far
+      end
+
+      state :root do
+        mixin :basics
+
+        # PROC definitions
+        rule %r((proc)(\s+)(\w+))ix do |m|
+          @proc_name = m[0].split(' ')[1].downcase
+          puts "    proc name: #{@proc_name}" if @debug
+          if self.class.sas_proc_names.include? @proc_name.upcase
+            groups Keyword, Text, Keyword
+          else
+            groups Keyword, Text, Name
+          end
+        
+          push :proc
+        end
+
+        # Data step definitions
+        rule /(data)(\s+)([\w\d\.]+)/i do
+          groups Keyword, Text, Name::Variable
+        end
+        # Libname definitions
+        rule /(libname)(\s+)([\w\d]+)/i do
+          groups Keyword, Text, Name::Variable
+        end
+
+        rule /\w[\w\d]*/ do |m|
```

Comment:
I see many uses of including both `\w` and `\d` in the same character set. That is unnecessary as `\d` is a subset of `\w`
`/\w+/` will match `foo` as well as `f00`.

You may test this by invoking `irb` with the `-w` flag:
```sh
irb -w
```

Reply:
Thank you. I have changed in commit https://github.com/rouge-ruby/rouge/pull/1107/commits/b1f3ae3e67387bc6750aa1eac34228c3a6dd5d04.